test/lit/passes/Oz.wast

```diff
@@ -5,14 +5,13 @@
 
 (module
   (memory 100 100)
-  ;; CHECK:      (rec
-  ;; CHECK-NEXT:  (type $i32_i32_=>_i32 (func (param i32 i32) (result i32)))
```

Comment:
I have no idea why this changed. @tlively is it something obvious perhaps? If not I can investigate.

Reply:
So what seems to be happening is that the old code would run GTO which would rewrite types (it modifies types "in place" by creating copies and modifying them, then switching everything to the new types). Doing that in isorecursive mode leads to the creation of a new rec group, and we end up printing it. But the new code avoids running that pass, so we don't add a rec group, and I'm guessing the printer will by default not emit a rec group at all if all we have is a single big one?

If so then this change seems expected and ok. Actually the old behavior of running some opt passes and getting a rec group printed out, when there wasn't one before, seems potentially confusing. (That makes me think again about the connection between closed-world and isorecursive types...)

Reply:
> I'm guessing the printer will by default not emit a rec group at all if all we have is a single big one?

Not quite. The printer will not emit a rec group for singleton rec groups, i.e. rec groups with just one element. The input module only has singleton rec groups, but `TypeUpdater` always creates single large rec groups, which is why the output has a rec group.

Reply:
Got it, thanks! So any use of TypeUpdater will go from many rec groups to a single one, which is then printed...

But isn't that unsafe in isorecursive mode? The many rec groups could have matched something in another module, but when we switch to a single big one they no longer do?

Reply:
It's unsafe with an open world, but fine with a closed world + isorecursive types.

Reply:
Ok, now I see. Then I think we will be doing the right thing after this PR, as we won't run those passes in an open world.

But is there any reason for someone to use closed world + isorecursive? What I mean is, if you are closed world, wouldn't you also want to use nominal? It's strictly better and has no downside AFAICT.

Reply:
Talked offline and decided that supporting all combinations of nominal or isorecursive and open or closed world doesn't make sense. The solution will be to delete the nominal type system and determine which optimizations are safe to run just based on the open/closed world flags.